### PR TITLE
feat(data-frame): Data Frame data source displays associated Workspace

### PR DESCRIPTION
**Pull request by cameronwaterman**
# Pull Request

## 🤨 Rationale

https://dev.azure.com/ni/DevCentral/_workitems/edit/2527389

## 👩‍💻 Implementation

Show Data Table Workspace name as `description` text is Data Table selection drop-down and move the ID to `title` (hover text). This involved needing to migrate the Data Table data source to extend `DataSourceBase`.

## 🧪 Testing

Tests continue to pass.

## ✅ Checklist

- [X] This PR has a title that follows the [commit message format](https://github.com/ni/systemlink-grafana-plugins#commit-message-format).

### CODE REVIEW

File: src/core/DataSourceBase.ts [hunk 1]
```diff
@@ -46,8 +47,8 @@ export abstract class DataSourceBase<TQuery extends DataQuery> extends DataSourc
     }
   }
 
-  get<T>(url: string) {
-    return this.fetch<T>({ method: 'GET', url });
+  get<T>(url: string, params: Record<string, any> | undefined = undefined) {
```
Comment (mure) [inline src/core/DataSourceBase.ts:50]: This is equivalent but shorter:
```suggestion
  get<T>(url: string, params?: Record<string, any>) {
```

File: src/datasources/data-frame/components/DataFrameQueryEditor.tsx [hunk 2]
```diff
@@ -40,10 +41,15 @@ export const DataFrameQueryEditor = (props: Props) => {
   };
 
   const loadTableOptions = _.debounce((query: string, cb?: LoadOptionsCallback<string>) => {
-    datasource
+    const tablesPromise = datasource
```
Comment (mure) [inline src/datasources/data-frame/components/DataFrameQueryEditor.tsx:44]:
```
Promise.all([datasource.queryTables(query), datasource.getWorkspaces()])
  .then(([tables, workspaces]) => cb?.(tables.map((t) => ({ label: t.name, value: t.id, title: t.id, description: getWorkspaceName(workspaces, t.workspace) }))))
  .catch(handleError);
```

Comment (mure) [reply src/datasources/data-frame/components/DataFrameQueryEditor.tsx:44]: I think this works the same

**Review by mure — APPROVED** [review]
Thanks for doing the refactor to extend from DataSourceBase. Glad to see that we didn't have to modify the tests at all.

File: src/core/DataSourceBase.ts (hunk 1 above)
Comment (cameronwaterman) [reply src/core/DataSourceBase.ts:50]: I could've swore I had this initially and my editor was complaining about it. Seems to be working now.

File: src/datasources/data-frame/components/DataFrameQueryEditor.tsx (hunk 2 above)
Comment (cameronwaterman) [reply src/datasources/data-frame/components/DataFrameQueryEditor.tsx:44]: Much cleaner - thanks!

**Pull request merged**